packages/headless/src/features/generated-answer/generated-answer-request.test.ts

```diff
@@ -147,4 +160,89 @@ describe('constructAnswerAPIQueryParams', () => {
       expectedStreamAnswerAPIParamWithDifferentFacetTimes
     );
   });
+
+  it('should include empty string referrer when there is NO referrer', () => {
+    const navigatorContext = buildMockNavigatorContextProvider();
+    const context = navigatorContext();
+    context.referrer = null;
+
+    const queryParams = constructAnswerAPIQueryParams(
+      streamAnswerAPIStateMock,
+      context
+    );
+
+    expect(queryParams).toEqual(expectedStreamAnswerAPIParamWithoutReferrer);
+  });
+
+  it('should include actionsHistory when analytics is enabled and history is present', () => {
+    const history = [
+      {name: 'search', value: 'some query', time: new Date().toISOString()},
+      {name: 'click', value: 'some uri', time: new Date().toISOString()},
+    ];
+    const mockHistoryStore: Pick<HistoryStore, 'getHistory'> = {
+      getHistory: vi.fn(() => history),
+    };
+    vi.spyOn(HistoryStore, 'getInstance').mockReturnValue(
+      mockHistoryStore as HistoryStore
+    );
+
+    const queryParams = constructAnswerAPIQueryParams(
+      streamAnswerAPIStateMockWithAnalyticsEnabled,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams.actionsHistory).toEqual(history);
+    expect(queryParams.actionsHistory?.length).toBe(2);
+  });
+
+  it('should include folding params when folding is disabled', () => {
+    const queryParams = constructAnswerAPIQueryParams(
+      streamAnswerAPIStateMockWithFoldingDisabled,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithFoldingDisabled
+    );
```

Comment:
I'm personally not a huge fan of this approach when testing the query parameters.

For instance in this test, our specific goal is to ensure that the folding parameters are set to certain values when folding is disabled. Instead of recursively checking for complete object equality with `toEqual`, we could just check that the `queryParams` match the desired object:

```typescript
expect(queryParams).toMatchObject({
  filterField: '',
  childField: '',
  parentField: '',
  filterFieldRange: 0,
});
```

I personally think this would make the tests easier to read. And also, each test would be testing only what it needs to test about the query parameters.